sormas-rest/src/test/java/de/symeda/sormas/rest/ExternalVisitsResourceTest.java

```diff
@@ -0,0 +1,150 @@
+		List<String> enumNames = Arrays.asList(
+			"JournalPersonDto",
+			"PersonSymptomJournalStatusDto",
+			"SymptomJournalStatus",
+			"ExternalVisitDto",
+			"Disease",
+			"VisitStatus",
+			"SymptomsDto",
+			"SymptomState",
+			"YesNoUnknown",
+			"TemperatureSource");
+		ArrayList releasedDetailList = new ArrayList();
+		ArrayList newDetailList = new ArrayList();
+
+		for (String name : enumNames) {
+			extractDetail(releasedSwaggerDocuMap, name, releasedDetailList);
+			extractDetail(newSwaggerDocuMap, name, newDetailList);
+		}
+		assertEquals(releasedDetailList, newDetailList);
+	}
+
+	/**
+	 * 
+	 * @param topLevelMap
+	 *            Nested Map from which to extract the information. It's supposed to be a mapped swagger.json
+	 * @param controller
+	 *            The name of the controller, e.g. External Visits Controller
+	 * @param list
+	 *            Extracted information is stored in this list
+	 * @return Documentation about any path found for the specified controller (e.g. /visits-external/person/{personUuid} for the External
+	 *         Visits Controller). This includes parameter names for that path, but not information about related enums.
+	 */
+	private static ArrayList extractPathsOfController(Map topLevelMap, String controller, ArrayList list) {
+		for (Object firstLayerKey : topLevelMap.keySet()) {
+			Object firstLayerValue = topLevelMap.get(firstLayerKey);
+			if (firstLayerValue instanceof Map) {
+				Map<String, Object> secondLayerMap = (Map<String, Object>) topLevelMap.get(firstLayerKey);
+				for (Object secondLayerKey : secondLayerMap.keySet()) {
+					Object secondLayerValue = secondLayerMap.get(secondLayerKey);
+					if (secondLayerValue instanceof Map) {
+						Map<String, Object> thirdLayerMap = (Map<String, Object>) secondLayerMap.get(secondLayerKey);
+						for (Object thirdLayerKey : thirdLayerMap.keySet()) {
+							// tags are always represented in the third layer and as ArrayLists
+							if ("tags".equals(thirdLayerKey) && thirdLayerMap.get(thirdLayerKey) instanceof ArrayList) {
+								ArrayList tags = (ArrayList) thirdLayerMap.get(thirdLayerKey);
+								if (tags.contains(controller)) {
+									list.add(firstLayerKey);
+									list.add(topLevelMap.get(firstLayerKey));
+								}
+							}
+						}
+					}
+				}
+				extractPathsOfController((Map) firstLayerValue, controller, list);
+			}
+		}
+		return list;
+	}
+
+	/**
+	 * 
+	 * @param topLevelMap
+	 *            Nested Map from which to extract the information. It's supposed to be a mapped swagger.json
+	 * @param detailName
+	 *            The name of the detail, e.g. JournalPersonDto.
+	 * @param list
+	 *            Extracted information is stored in this list
+	 * @return
+	 *         Documentation found about the detail. The Map is searched for a key equal to detailName, an it, plus the according value is
+	 *         added to the list.
+	 */
+	private static ArrayList extractDetail(Map topLevelMap, String detailName, ArrayList list) {
```

Comment:
same as above

Reply:
Also same as above. At least, the code here is simpler.